twisted/python/threadpool.py

```diff
@@ -209,6 +210,25 @@ def stop(self):
             thread.join()
 
 
+    @defer.inlineCallbacks
+    def stopAndWait(self):
+        """
+        Post stop request to all working threads and wait for them to finish.
+
+        Returns a deferred which fires when all workers are terminated.
+        """
+        self.joined = True
+
+        while self.workers:
+            self.q.put(WorkerStop)
+            self.workers -= 1
+
+        for thread in copy.copy(self.threads):
+            yield threads.deferToThread(thread.join)
+
+        defer.returnValue(None)
```

Comment:
since we don't care about the return value, you can just ignore it, as this is a return on the last line and returnValue(None) is done by default by the inlineCallbacks